util/chplenv/chplenv_verify.py

```diff
@@ -0,0 +1,289 @@
+        with open(hello_file[1], "w") as f:
+            f.write(self._program())
+
+        with FileCleanupManager(tmpdir) as fcm:
+            cmd = self._compiler() + [hello_file[1], "-o", f"{tmpdir}/a.out"] + self._compiler_args()
+            self.log("Trying to compile with '{}'".format(" ".join(cmd)))
+            succ, ret, out, _ = try_run_command(cmd, combine_output=True)
+            if not succ or ret != 0:
+                out = out if out is not None else "unknown error"
+                self.msg = "Failed to compile program: {}".format(out)
+                return False
+
+            # Run the program
+            cmd = [f"{tmpdir}/a.out"]
+            self.log("Trying to run with '{}'".format(" ".join(cmd)))
+            succ, ret, out, _ = try_run_command(cmd, combine_output=True)
+            if not succ or ret != 0:
+                out = out if out is not None else "unknown error"
+                self.msg = "Failed to run program: {}".format(out)
+                return False
+
+        return True
+
+    def explain(self):
+        if not self.msg:
+            compiler = self._compiler()
+            if not compiler:
+                self.msg = "No compiler found"
+            self.msg = "{} compiler '{}' cannot compile programs".format(self._lang(), compiler[0])
+        return super().explain()
+
+
+@register
+class TestHostCompile(TestCompile):
+    """
+    Try and compile a hello world program using CHPL_HOST_CXX
+    """
+
+    def _suffix(self):
+        return ".cpp"
+
+    def _lang(self):
+        return "C++"
+
+    def _compiler(self):
+        cxx = self.chplenv.get("CHPL_HOST_CXX")
+        if not cxx:
+            self.msg = "CHPL_HOST_CXX is missing"
+            return []
+        return shlex.split(cxx)
+
+    def _program(self):
+        return """
+#include <iostream>
+int main() {
+  std::cout << "Hello, World!" << std::endl;
+  return 0;
+}
+"""
+
+@register(after=TestHostCompile)
+class TestTargetCompileCC(TestCompile):
+    """
+    Try and compile a hello world program using CHPL_TARGET_CC
+    """
+
+    def _suffix(self):
+        return ".c"
+
+    def _lang(self):
+        return "C"
+
+    def _compiler(self):
+        cc = self.chplenv.get("CHPL_TARGET_CC")
+        if not cc:
+            self.msg = "CHPL_TARGET_CC is missing"
+            return []
+        return shlex.split(cc)
+
+    def _program(self):
+        return """
+#include <stdio.h>
+int main() {
+    printf("Hello, World!\\n");
+    return 0;
+}
+"""
+
+
+@register(after=TestTargetCompileCC)
+class TestTargetCompileCXX(TestCompile):
+    """
+    Try and compile a hello world program using CHPL_TARGET_CXX
+    """
+
+    def _suffix(self):
+        return ".cpp"
+
+    def _lang(self):
+        return "C++"
+
+    def _compiler(self):
+        cxx = self.chplenv.get("CHPL_TARGET_CXX")
+        if not cxx:
+            self.msg = "CHPL_TARGET_CXX is missing"
+            return []
+        return shlex.split(cxx)
```

Comment:
All of these are the same. Why not define a `_compiler_env_var` per-child, and move this logic into the parent class?